pallets/merkle-airdrop/src/lib.rs

```diff
@@ -0,0 +1,617 @@
+            let amount_bytes = amount.encode();
+
+            // Concatenate the bytes
+            let combined = [account_bytes.as_slice(), amount_bytes.as_slice()].concat();
+
+            blake2_256(&combined)
+        }
+
+        /// Calculates the parent hash in a Merkle tree using Blake2.
+        ///
+        /// This function combines two child hashes to create their parent hash in the Merkle tree.
+        /// The children are ordered lexicographically before hashing to ensure consistency.
+        ///
+        /// # Parameters
+        ///
+        /// * `left` - The first child hash
+        /// * `right` - The second child hash
+        ///
+        /// # Returns
+        ///
+        /// A 32-byte array containing the Blake2 hash of the combined children
+        pub fn calculate_parent_hash_blake2(left: &[u8; 32], right: &[u8; 32]) -> [u8; 32] {
+            // Ensure consistent ordering of inputs (important for verification)
+            let combined = if left < right {
+                [left.as_slice(), right.as_slice()].concat()
+            } else {
+                [right.as_slice(), left.as_slice()].concat()
+            };
+
+            blake2_256(&combined)
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Create a new airdrop with a Merkle root.
+        ///
+        /// The Merkle root is a cryptographic hash that represents all valid claims
+        /// for this airdrop. Users will later provide Merkle proofs to verify their
+        /// eligibility to claim tokens.
+        ///
+        /// # Parameters
+        ///
+        /// * `origin` - The origin of the call (must be signed)
+        /// * `merkle_root` - The Merkle root hash representing all valid claims
+        ///
+        /// # Errors
+        ///
+        /// * `TooManyAirdrops` - If the maximum number of airdrops has been reached
+        /// * `AirdropAlreadyExists` - If an airdrop with this ID already exists
+        #[pallet::call_index(0)]
+        #[pallet::weight(T::WeightInfo::create_airdrop())]
+        pub fn create_airdrop(origin: OriginFor<T>, merkle_root: MerkleRoot) -> DispatchResult {
+            log::info!(
+                target: "merkle-airdrop",
+                "🌟 create_airdrop called with root: {:?}",
+                merkle_root
+            );
+
+            let who = ensure_signed(origin)?;
+            log::info!(
+                target: "merkle-airdrop",
+                "✅ Caller: {:?}",
+                who
+            );
+
+            // Get the next available airdrop ID
+            let airdrop_id = Self::next_airdrop_id();
+            log::info!(
+                target: "merkle-airdrop",
+                "📊 Next airdrop ID: {:?}",
+                airdrop_id
+            );
+
+            // Ensure we haven't reached the maximum number of airdrops
+            ensure!(
+                airdrop_id < T::MaxAirdrops::get(),
+                Error::<T>::TooManyAirdrops
```

Comment:
I think this implementation caps the total number of airdrops that will ever happen. Can we just drop MaxAirdrops as a concept completely? @dastansam ?

Reply:
there's no good way to tell when an airdrop is finished, unfortunately. we could say when it gets depleted we could delete it, but the current design allows for an airdrop to be funded multiple times. We could allow airdrops to be deleted by someone if their balance is zero maybe?

Reply:
I have implemented delete airdrop here: https://github.com/Quantus-Network/chain/pull/10/commits/59321021c7edbae6cde46ae691eb4ee723691773
I am not sure those to which accounts to allow this action, so even if we can configure it, currently I allow it to all signed origins